rocAL/include/readers/image/image_reader.h

```diff
@@ -171,6 +171,10 @@ class Reader {
     //! Returns the name/identifier of the last item opened in this resource
     virtual std::string id() = 0;
     //! Returns the number of items remained in this resource
+
+     //! Returns the path of the last item opened in this resource
+    virtual std::string file_path() { return {}; }
```

Comment:
why is this returning empty string which can lead to issues

Reply:
This is a virtual function, hence we are returning an empty string. Currently it is used in file source reader only, since we did not want to add the API to all the readers which would increase number of file changes